### PR TITLE
Fix: avoid duplicate loop key in err handler search

### PR DESCRIPTION
**Pull request by laggardkernel**
Very minimal change. The code is self-explanation. `code` in `for c in [code, None]` may be `None`. Made a deduplication before do the loop.

Checklist:

- [ ] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the docs folder and in code.
- [ ] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [ ] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

File: src/flask/app.py [hunk 1]
```diff
@@ -1276,7 +1276,7 @@ def _find_error_handler(self, e: Exception) -> t.Optional[ErrorHandlerCallable]:
         """
         exc_class, code = self._get_exc_class_and_code(type(e))
 
-        for c in [code, None]:
+        for c in [None] if code is None else [code, None]:
```
Comment (pgjones) [inline src/flask/app.py:1279]: What about `for c in {code, None}:`? Reads more clearly clearer to me

Comment (davidism) [reply src/flask/app.py:1279]: Order isn't guaranteed for sets.

Comment (pgjones) [reply src/flask/app.py:1279]: Oh yea, the order matters.

**Comment by laggardkernel** [comment]
Adjusted the code a little to avoid typing error, and rebased onto the HEAD of branch '2.0.x'.

**Pull request merged**